diff --git a/Makefile.am b/Makefile.am
index d31395772..1b4f0443b 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1105,6 +1105,7 @@ libsss_sbus_la_LIBADD = \
     $(TALLOC_LIBS) \
     $(TEVENT_LIBS) \
     $(DBUS_LIBS) \
+    $(UNICODE_LIBS) \
     $(NULL)
 libsss_sbus_la_CFLAGS = \
     $(AM_CFLAGS) \
@@ -1146,6 +1147,7 @@ libsss_sbus_sync_la_CFLAGS = \
     $(AM_CFLAGS) \
     $(TALLOC_CFLAGS) \
     $(DBUS_CFLAGS) \
+    $(UNICODE_LIBS) \
     $(NULL)
 libsss_sbus_sync_la_LDFLAGS = \
     -avoid-version \
@@ -1209,28 +1211,6 @@ libsss_iface_sync_la_LDFLAGS = \
     -avoid-version \
     $(NULL)

-if BUILD_WITH_LIBSECRET
-pkglib_LTLIBRARIES += libsss_secrets.la
-
-libsss_secrets_la_SOURCES = \
-    src/util/secrets/secrets.c \
-    src/util/secrets/config.c \
-    $(NULL)
-libsss_secrets_la_CFLAGS = \
-    $(AM_CFLAGS) \
-    $(NULL)
-libsss_secrets_la_LIBADD = \
-    $(TALLOC_LIBS) \
-    $(LDB_LIBS) \
-    libsss_crypt.la \
-    libsss_debug.la \
-    libsss_util.la \
-    $(NULL)
-libsss_secrets_la_LDFLAGS = \
-    -avoid-version \
-    $(NULL)
-endif
-
 pkglib_LTLIBRARIES += libsss_util.la
 libsss_util_la_SOURCES = \
     src/confdb/confdb.c \
@@ -1314,6 +1294,27 @@ libsss_util_la_LIBADD += stap_generated_probes.lo
 endif
 libsss_util_la_LDFLAGS = -avoid-version

+if BUILD_WITH_LIBSECRET
+pkglib_LTLIBRARIES += libsss_secrets.la
+libsss_secrets_la_SOURCES = \
+    src/util/secrets/secrets.c \
+    src/util/secrets/config.c \
+    $(NULL)
+libsss_secrets_la_CFLAGS = \
+    $(AM_CFLAGS) \
+    $(NULL)
+libsss_secrets_la_LIBADD = \
+    $(TALLOC_LIBS) \
+    $(LDB_LIBS) \
+    libsss_crypt.la \
+    libsss_debug.la \
+    libsss_util.la \
+    $(NULL)
+libsss_secrets_la_LDFLAGS = \
+    -avoid-version \
+    $(NULL)
+endif
+
 pkglib_LTLIBRARIES += libsss_semanage.la
 libsss_semanage_la_CFLAGS = \
     $(AM_CFLAGS) \
